### PR TITLE
Enhance scene tabs

### PR DESCRIPTION
**Pull request by volzhs**
- show scene thumbnail on hover
- resize if has many tabs
- show full scene file name with current edited scene
- can be customized EditorSettings > Interface > Scene Tab
- close scene with mouse middle button

scene tab 5

### CODE REVIEW

**Comment by nunodonato** [comment]
awesome work, as alwasy :D 
is the thumbnail toggable? can the delay be changed?

**Comment by volzhs** [comment]
@nunodonato yes. it can be toggle in editor settings. but there is nothing about delay.

**Comment by Zylann** [comment]
It would be nice to also have middle-click close and a hover effect, but that's awesome already :)

**Comment by volzhs** [comment]
@Zylann oh, I'd thought about that too, but forgot. I will check it later.

File: editor/editor_data.cpp [hunk 1]
```diff
@@ -675,7 +675,12 @@ String EditorData::get_scene_title(int p_idx) const {
 		return "[empty]";
 	if (edited_scene[p_idx].root->get_filename() == "")
 		return "[unsaved]";
-	return edited_scene[p_idx].root->get_filename().get_file();
+	bool show_ext = EDITOR_DEF("interface/scene_tab/show_extension", false);
+	String name = edited_scene[p_idx].root->get_filename().get_file();
+	if (!show_ext) {
+		name = name.split(".")[0];
```
Comment (akien-mga) [inline editor/editor_data.cpp]: That won't work if the file is name `some.scene.tscn` :)

Comment (volzhs) [inline editor/editor_data.cpp]: @akien-mga oh my... right. will fix it.

Comment (akien-mga) [inline editor/editor_data.cpp]: You'd have more luck with http://docs.godotengine.org/en/latest/classes/class_string.html#class-string-get-basename :)

File: editor/editor_data.cpp [hunk 2]
```diff
@@ -675,7 +675,12 @@ String EditorData::get_scene_title(int p_idx) const {
 		return "[empty]";
 	if (edited_scene[p_idx].root->get_filename() == "")
 		return "[unsaved]";
-	return edited_scene[p_idx].root->get_filename().get_file();
+	bool show_ext = EDITOR_DEF("interface/scene_tab/show_extension", false);
```
Comment (akien-mga) [inline editor/editor_data.cpp]: Maybe `interface/scene_tabs` (with an s) would be a better name?

Comment (volzhs) [inline editor/editor_data.cpp]: will change that too. :) tomorrow...

File: editor/editor_data.cpp (hunk 1 above)
Comment (volzhs) [inline editor/editor_data.cpp]: @akien-mga I totally forgot that. thanks. :)

**Comment by volzhs** [comment]
@Zylann closing scene with mouse middle button is just added. :)
@akien-mga updated commit.

**Comment by karroffel** [comment]
When you change the theme in the editor the tabs stay the same, you need to restart the editor.

Was this always the case or did this PR cause this?

image

**Comment by kubecz3k** [comment]
Wow, it's awesome! Would be great if we could reposition tabs as well :D

**Comment by volzhs** [comment]
@karroffel I will check that.
@kubecz3k I checked about re-positioning tabs, but it was too complicated.

**Comment by kubecz3k** [comment]
@volzhs I would want to implement tab repositioning, would be cool if you could drop me some tips :)

**Pull request merged**